server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/test/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutesTest.java

```diff
@@ -112,6 +129,1083 @@ void afterEach() {
+                    .get(taskId + "/await")
+                .then()
+                    .body("status", is("completed"));
+
+                assertThat(hasAnyMail(USER)).isFalse();
+            }
+
+            @Test
+            void restoreShouldAppendMessageToMailboxWhenMatchingSubjectEquals() throws Exception {
+                vault.append(USER, FINAL_STAGE.get()
+                    .subject("subject should match")
+                    .build(), new ByteArrayInputStream(CONTENT)).block();
+
+                String query = "{" +
+                    "  \"criterions\": [" +
+                    "    {" +
+                    "      \"fieldName\": \"subject\"," +
+                    "      \"operator\": \"contains\"," +
+                    "      \"testedValue\": \"subject should match\"" +
+                    "    }" +
+                    "  ]" +
+                    "}";
+
+                String taskId =
+                    given()
+                        .body(query)
+                        .when()
+                        .post(USER.asString())
+                        .jsonPath()
+                        .get("taskId");
+
+                given()
+                    .basePath(TasksRoutes.BASE)
+                    .when()
+                    .get(taskId + "/await")
+                    .then()
+                    .body("status", is("completed"));
+
+                assertThat(restoreMessageContents(USER))
+                    .hasSize(1)
+                    .hasOnlyOneElementSatisfying(messageIS -> assertThat(messageIS).hasSameContentAs(new ByteArrayInputStream(CONTENT)));
+            }
+
+            @Test
+            void restoreShouldNotAppendMessageToMailboxWhenSubjectDoesntEquals() throws Exception {
+                vault.append(USER, FINAL_STAGE.get()
+                    .subject("subject")
+                    .build(), new ByteArrayInputStream(CONTENT)).block();
+
+                String query = "{" +
+                    "  \"criterions\": [" +
+                    "    {" +
+                    "      \"fieldName\": \"subject\"," +
+                    "      \"operator\": \"equals\"," +
+                    "      \"testedValue\": \"SUBJECT\"" +
+                    "    }" +
+                    "  ]" +
+                    "}";
+
+                String taskId =
+                    given()
+                        .body(query)
+                    .when()
+                        .post(USER.asString())
+                        .jsonPath()
+                        .get("taskId");
+
+                given()
+                    .basePath(TasksRoutes.BASE)
+                .when()
+                    .get(taskId + "/await")
+                .then()
+                    .body("status", is("completed"));
+
+                assertThat(hasAnyMail(USER)).isFalse();
+            }
+
+            @Test
+            void restoreShouldAppendMessageToMailboxWhenMatchingSubjectEqualsIgnore() throws Exception {
```

Comment:
Doesn't test more or less the same as `restoreShouldAppendMessageToMailboxWhenMatchingSubjectContainsIgnoreCase` ?

Reply:
I have a test for that!

Reply:
I meant those two tests were doing the same in my opinion, like a duplicate

Reply:
there are two test case for two different scenario, one is using `containsIgnoreCase` operator, one is using `equalsIngoreCase` operator. Then I need to write test for covering both cases. Although, input jsons look 99% the same except the operators